File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    const dict = {};
    tangerine.forEach((x)=>(dict[x] = (dict[x] || 0) + 1))
    const arr = Object.values(dict).sort((a,b)=>b-a)
    let sub = 0
    for (let el of arr){
        sub += el
        if (sub < k){
            answer += 1
        }
        else if (sub == k){
            answer += 1
            break
        }
        else {
            answer += 1
            break
        }
    }    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        HashMap<Integer, Integer> tangCnt = new HashMap<Integer, Integer>();
        
        for (int size: tangerine) {
            int cnt = tangCnt.getOrDefault(size, 0);
            tangCnt.put(size, cnt + 1);
        }
        List<Map.Entry<Integer, Integer>> entries = new ArrayList<>(tangCnt.entrySet());
        entries.sort((e1, e2) -> Integer.compare(e2.getValue(), e1.getValue()));
        
        for (Map.Entry<Integer, Integer> entry: entries) {
            if (k <= 0) break;
            k -= entry.getValue();
            answer++;
        }
        
        return answer;
    }
}